### PR TITLE
[addons] fix addon install process percent indication

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description
Previously, calling CAddonInstaller :: OnJobProgress always assumed it
was 100% Total, but this is not really what it is like, as in the
example 0-4.

With this commit the total value is calculated to 100%.

## Motivation and Context
Make the Addon System better :-)

## How Has This Been Tested?
See screenshots

## Screenshots (if appropriate):
Before:
bildschirmfoto von 2018-11-16 00-04-25

After:
bildschirmfoto von 2018-11-16 00-09-05
bildschirmfoto von 2018-11-23 19-09-43

(see the Downloading on the right)

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by AlwinEsch** [comment]
Added a second commit, which sets the text to a more accurate description.

I'm not sure if this is OK for Leia. Also, I am not sure if the code type is so OK.

**Comment by MartijnKaijser** [comment]
If it get approval from another dev it can be included

File: xbmc/addons/AddonInstaller.h [hunk 1]
```diff
@@ -173,6 +175,7 @@ class CAddonInstallJob : public CFileOperationJob
   ADDON::RepositoryPtr m_repo;
   bool m_isUpdate;
   bool m_isAutoUpdate;
+  const char* m_currentType = "";
```
Comment (Rechi) [inline xbmc/addons/AddonInstaller.h]: make this an enum, don't abuse string compares for state describing

Comment (AlwinEsch) [reply xbmc/addons/AddonInstaller.h]: There you right. 
Just used it as it is in its parent (https://github.com/xbmc/xbmc/blob/master/xbmc/utils/Job.h#L137).

I look then to use a:
```cpp
void CAddonInstaller::OnJobComplete(unsigned int jobID, bool success, CJob* job)
{
  CAddonInstallJob* addonJob = dynamic_cast<CAddonInstallJob*>(job);
```
To get another type of value.

Comment (Rechi) [reply xbmc/addons/AddonInstaller.h]: Haven't looked at the parent class, then add the following in this file and use those variables.

```
static constexpr char* TYPE_DOWNLOAD = "DOWNLOAD";
static constexpr char* TYPE_INSTALL = "INSTALL";
```

File: xbmc/addons/AddonInstaller.h [hunk 2]
```diff
@@ -100,6 +100,7 @@ class CAddonInstaller : public IJobCallback
     }
     unsigned int jobID;
     unsigned int progress;
+    bool downloadFinshed;
```
Comment (Rechi) [inline xbmc/addons/AddonInstaller.h]:
```suggestion
    bool downloadFinshed = false;
```

**Comment by AlwinEsch** [comment]
Thanks @Rechi :+1: , is updated

**Comment by Rechi** [comment]
wrong revision was build
jenkins build this please

File: xbmc/addons/AddonInstaller.h [hunk 3]
```diff
@@ -146,6 +144,16 @@ class CAddonInstallJob : public CFileOperationJob
 
   bool DoWork() override;
 
+  static constexpr const char* TYPE_DOWNLOAD = "DOWNLOAD";
+  static constexpr const char* TYPE_INSTALL = "INSTALL";
+  /*!
+   * \brief Returns the current processing type in the installation job
+   *
+   * \return The cuurent processing type as string, can be \ref TYPE_DOWNLOAD or
```
Comment (MartijnKaijser) [inline xbmc/addons/AddonInstaller.h]: cuurent -> current

Comment (AlwinEsch) [reply xbmc/addons/AddonInstaller.h]: Upps, Sorry!

**Comment by AlwinEsch** [comment]
jenkins build this please

**Pull request merged**